src/json.js

```diff
@@ -164,108 +40,3 @@ export function getChildJsonConfig(element) {
     throw new Error('Failed to parse <script> contents. Is it valid JSON?');
   }
 }
-
-/**
- * Deeply checks strict equality of items in nested arrays and objects.
- *
- * @param {JSONValueDef} a
- * @param {JSONValueDef} b
- * @param {number} depth The maximum depth. Must be finite.
- * @return {boolean}
- * @throws {Error} If depth argument is not finite.
- */
-export function deepEquals(a, b, depth = 5) {
-  if (!isFinite(depth) || depth < 0) {
-    throw new Error('Invalid depth: ' + depth);
-  }
-  if (a === b) {
-    return true;
-  }
-  /** @type {!Array<{a: JSONValueDef, b: JSONValueDef, depth: number}>} */
-  const queue = [{a, b, depth}];
-  while (queue.length > 0) {
-    const {a, b, depth} = queue.shift();
-    // Only check deep equality if depth > 0.
-    if (depth > 0) {
-      if (typeof a !== typeof b) {
-        return false;
-      } else if (Array.isArray(a) && Array.isArray(b)) {
-        if (a.length !== b.length) {
-          return false;
-        }
-        for (let i = 0; i < a.length; i++) {
-          queue.push({a: a[i], b: b[i], depth: depth - 1});
-        }
-        continue;
-      } else if (a && b && typeof a === 'object' && typeof b === 'object') {
-        const keysA = Object.keys(/** @type {!Object} */ (a));
-        const keysB = Object.keys(/** @type {!Object} */ (b));
-        if (keysA.length !== keysB.length) {
-          return false;
-        }
-        for (let i = 0; i < keysA.length; i++) {
-          const k = keysA[i];
-          queue.push({a: a[k], b: b[k], depth: depth - 1});
-        }
-        continue;
-      }
-    }
-    // If we get here, then depth == 0 or (a, b) are primitives.
-    if (a !== b) {
-      return false;
-    }
-  }
-  return true;
-}
-
-/**
- * @param {*} obj
- * @param {string} key
- * @return {boolean}
- */
-function hasOwnProperty(obj, key) {
-  if (obj == null || typeof obj != 'object') {
```

Comment:
I'm check that this change was not preserved. Particularly, I think something like `deepEquals(null, {})` will now throw?

Reply:
`deepEquals` doesn't call this. It was called by `recreateNonProtoObject` and `getValueForExpr` (now in `core/types/object/index.js`)
- `recreateNonProtoObject` deals with a `!JsonObject` (inside a for...of loop) so it can call `hasOwn` without checking that it's an object
- `getValueForExpr` calls it within an existing conditional that asserts truthiness (ie. != null) and I added a `typeof` check to assert object-ness

There were no other callers, and it's not exported, so it didn't seem worth keeping

Reply:
![image](https://user-images.githubusercontent.com/6694512/118176952-10d00500-b400-11eb-8add-91985105a536.png)
![image](https://user-images.githubusercontent.com/6694512/118176984-1cbbc700-b400-11eb-9cdc-959e546eba7d.png)